client/signup/controller.js

```diff
@@ -135,16 +139,35 @@ export default {
 
 				if (
 					existingUsersOnboardingVariant === 'treatment' ||
-					[ 'en', 'en-gb' ].includes( locale )
+					[ 'en', 'en-gb' ].includes( userLocale )
 				) {
-					gutenbergRedirect( context.params.flowName );
+					gutenbergRedirect();
 					return;
 				}
 			}
 
 			waitForHttpData( () => ( { geo: requestGeoLocation() } ) )
 				.then( ( { geo } ) => {
 					const countryCode = geo.data;
+					const localeFromParams = context.params.lang;
+
+					if (
+						( countryCode !== 'US' || ( localeFromParams && localeFromParams !== 'en' ) ) &&
+						! userLoggedIn &&
+						flowName === 'free'
+					) {
+						// Assign to the experiment only non EN-US, logged-in users, creating a site using 'free' flow.
+						const newUsersFreePlanOnboardingVariant = getVariationForUser(
+							state,
+							'new_onboarding_free_plan_users_v2'
+						);
```

Comment:
This is always `NULL` for new users as confirmed with @jessie-ross since the fetching (assigning to user to experiment) doesn't happen until `<QueryExperiments />` is mounted. Examples are in `client/layout/index.jsx` or
`client/signup/steps/plans/index.jsx`.